core/src/main/java/de/javakaffee/web/msm/StorageClientFactory.java

```diff
@@ -43,6 +43,15 @@ MemcachedClient createCouchbaseClient(MemcachedNodesManager memcachedNodesManage
                 long maxReconnectDelay, Statistics statistics );
     }
 
+    static interface MemcachedNodeURLBasedConnectionFactory {
+        ConnectionFactory createBinaryConnectionFactory(long operationTimeout,
+                                                        long maxReconnectDelay, boolean isClientDynamicMode);
```

Comment:
Normally the client mode is determined by the node name in aws client.  If the node name contains ".cfg." it would be a dynamic mode. Either we pass the Boolean or we pass the node name?  
Check the implementation in
aws-elasticache-cluster-client-memcachedfor-java --> 
/src/main/java/net/spy/memcached/MemcachedClient.java line 
...
 if(determineClientMode){
      if(addrs.size() == 1){
        if(addrs.get(0) == null){
          throw new NullPointerException("Socket address is null");
        }
        String hostName = addrs.get(0).getHostName();
        //All config endpoints has ".cfg." subdomain in the DNS name.
        if(hostName != null && hostName.contains(".cfg.")){
          cf = new DefaultConnectionFactory(ClientMode.Dynamic);
        }
      }
      //Fallback to static mode
      if(cf == null){
        cf = new DefaultConnectionFactory(ClientMode.Static);
      }
...